scripts/setup-adapters.sh

```diff
@@ -20,6 +20,111 @@ SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
 source $SCRIPTDIR/setup-helper-functions.sh
 DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
 
+function install_gcs-sdk-cpp {
+  user=`whoami`
+  case "$user" in
+    root) echo 'Running as root' ; sudocmd="";;
+       *) echo 'Running as other user than root' ; sudocmd="sudo";;
+  esac
+  # install gcs dependencies
+  # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/packaging.md#required-libraries
+  # install abseil
+  echo " Installing abseil..."
+  mkdir -p ${DEPENDENCY_DIR}/abseil-cpp && cd ${DEPENDENCY_DIR}/abseil-cpp
+  curl -sSL https://github.com/abseil/abseil-cpp/archive/20211102.0.tar.gz | \
+      tar -xzf - --strip-components=1 && \
+      sed -i 's/^#define ABSL_OPTION_USE_\(.*\) 2/#define ABSL_OPTION_USE_\1 0/' "absl/base/options.h" && \
+      cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBUILD_TESTING=OFF \
+        -DBUILD_SHARED_LIBS=yes \
+        -DCMAKE_CXX_STANDARD=11 \
+        -H. -Bcmake-out && \
+      cmake --build cmake-out -- -j ${NCPU:-4} && \
+  $sudocmd cmake --build cmake-out --target install -- -j ${NCPU:-4} && \
+  $sudocmd ldconfig
+
+
+  # install protobuf
+  echo " Installing protobuf..."
```

Comment:
`protobuf` is already a dependency for Velox. The minimum is `3.0.0`. The documentation of gcs says the minimum needed is `3.21.1`. Is that accurate? Looks like even `3.20.1` works. Two protobuf installs to default locations might lead to issues.
What do you think of installing all the artifacts related to `gcs` to a folder say `gcs_artifacts` and then point CMake to this folder?

Reply:
@majetideepak I rebased my PR after the HDFS change and extended linux-parquet-hive-storage-adapters-build to include GCS as well. As for protobuf, I haven't seen any issue especially sinec 3.20.1 doesnt conflict with 3.0.0 so how about I uninstall (apt remove protobuf) before I install the 3.20.1

(I checked the 3.0.0 is not working for gcs)

Reply:
@ecopty I don't think uninstall is an option since there could be other external dependencies. Since the adapter dependencies are installed in the `DEPENDENCY_DIR`, it should be okay.
Thanks for all the changes. I will make a final pass tomorrow.